core/base/slot_chain.go

```diff
@@ -152,7 +161,7 @@ func (sc *SlotChain) Entry(ctx *EntryContext) *TokenResult {
 	if len(ss) > 0 {
 		for _, s := range ss {
 			// indicate the result of rule based checking slot.
-			if ruleCheckRet.Status() == ResultStatusPass {
+			if ruleCheckRet.IsPass() {
```

Comment:
Here we may need a convention: every final status indicates "pass", except the block status (and other special statuses), in case of bugs that yield alien results.

Reply:
So, if the token result is not blocked, the rest all are pass. Is it rigth?